internal/xdscache/v3/endpointstranslator.go

```diff
@@ -54,10 +54,13 @@ func RecalculateEndpoints(port, healthPort v1.ServicePort, eps *v1.Endpoints) []
 		}
 
 		for _, p := range s.Ports {
-			if (healthPort.Protocol != p.Protocol || port.Protocol != p.Protocol) && p.Protocol != v1.ProtocolTCP {
+			if p.Protocol != v1.ProtocolTCP {
 				// NOTE: we only support "TCP", which is the default.
 				continue
 			}
+			if healthPort.Protocol != p.Protocol || port.Protocol != p.Protocol {
+				continue
+			}
```

Comment:
Digging into the unit test failures, I think it's fine to drop this if statement here. This code is intended to just skip over any endpoints ports that are not using TCP protocol, since that's all we support. During the DAG build we already ensure that `healthPort.Protocol` and `port.Protocol` are TCP, so as long as we're filtering out endpoint ports with non-TCP protocol (which we're doing above), then this check on L61-63 is not needed.

Reply:
Yeah in practice we'll never create a cluster with a service port or health port that is not TCP (the `LookupService` helper prevents this), so it seems valid to drop this.

The tests are failing because we don't set a health port or even a port number in the `ServicePort` for most of them (and `dag.ServiceCluster.Services[].HealthPort` is an empty value of `corev1.ServicePort` rather than a nil pointer). In these tests as written `healthPort.Protocol` (empty) will never be equal to `p.Protocol` (TCP) and in many of them `port.Protocol` will also be empty. In practice we always have a full value set for `dag.ServiceCluster.Services[].HealthPort` (either set to the same as `ServicePort` or an independent valid value).

Reply:
I suggest also for readability, renaming `p` to `endpointPort` (or something similar, though `EndpointPort` is the type so seems like a good fit) and `port` to `servicePort`

Reply:
We could of course fix all the tests to fill in proper values everywhere to keep the original intent (health port and service port have to match the endpoint port protocol), and/or use an empty string check to make the tests happy (e.g. `if (healthPort.Protocol != "" && healthPort.Protocol != p.Protocol) || (servicePort.Protocol != "" && servicePort.Protocol != p.Protocol) { ... }`)